src/main/java/core/mate/academy/service/MachineProducer.java

```diff
@@ -1,11 +1,12 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Machine;
 import java.util.List;
 
-public interface MachineProducer {
+public interface MachineProducer<T extends Machine> {
     /**
      * In your implementations - create 2-3 models, add them to the list and return
      * @return - the list of models
      */
```

Comment:
The comment suggests that implementations should create 2-3 models and return them in the list. Ensure that the implementations of this interface, such as `BulldozerProducer` and `ExcavatorProducer`, follow this guideline by returning a list with 2-3 instances.